### DISCUSSION TITLE
current deseq2.r workflow error at line 190

### DISCUSSION THREAD

**Discussion by njbowen** [biostars/biostar-handbook-issues#291] [Q&A] [answered]
The command at line 189/190

# Rename the columns.
data = rename(data, PValue=pvalue, FDR=padj)

is returning this error:  Error: object 'PValue' not found

**Comment by ialbert** [comment]
how was this code obtained (which book is this from? RNA-seq by example or Biostar Workflows)

**Reply by njbowen** [reply]
Workflows,
Upgrade the bio package.
pip install bio --upgrade

Download the workflow source code.
bio code

**Comment by njbowen** [comment]
forgot to say, running this in an Rstudio script/markdown document

i substituted this: 
data = dplyr::rename(data, PValue=pvalue, FDR=padj)

so it's an issue with rename in the current environment that I am running RStudio in. 
so probably an error specific to my script and R libraries that are loaded, like some conflict in the R command rename between libraries.

**Reply by ialbert** [reply, marked as answer]
ah yes, one of the fundamental problems with R is that names leak out of scope, and it may overwrite one name with another

I'll look into it to make it more robust, thanks for the tip
